### ISSUE TITLE
🐛 Cursors use `SystemMouseCursors.click` instead of `SystemMouseCursors.basic`

### ISSUE THREAD

**Issue by wilsonowilson** [macosui/macos_ui#139]
**Description**

Clickable macOS components do not use the "click cursor". They have always used the default cursor. This forms a distinction between using the web on macOS and interacting with macOS applications.

So by default, clickable elements in macos_ui should use `SystemMouseCursors.basic` instead of `SystemMouseCursors.click`. I also believe an option should be provided to customize that.

**Comment by bdlukaa** [comment]
You can wrap the widget you want in a `MouseRegion` to add a mouse cursor.
```dart
MouseRegion(
  cursor: SystemMouseCursos.click,
  child: Checkbox(...),
),
```

**Comment by wilsonowilson** [comment]
Yes I'm aware of that, but that's not what I'm talking about.

This is a snippet from `back_button.dart`.
```dart
MouseRegion(
      cursor: SystemMouseCursors.click,
      child: GestureDetector(
        behavior: HitTestBehavior.opaque,
 ```
 I'm saying that the in-built widgets shouldn't use `SystemMouseCursors.click` by default.

**Comment by bdlukaa** [comment]
Will be fixed alongside https://github.com/GroovinChip/macos_ui/issues/142

**Comment by wilsonowilson** [comment]
The pull request doesn't address what I'm talking about. @GroovinChip might explain better.

**Comment by GroovinChip** [comment]
Wasn't it that mouse cursors should never be pointers? Material State Props can help us ensure that, right? Along with making it customizable?

**Comment by wilsonowilson** [comment]
It can ensure customisation yes, but it doesn't directly relate to the issue. I can open a separate PR once #142 is merged.

And I'm not sure we'd want to use `MacosStateProperty` on cursors. I haven't seen that usage even in the material buttons.

**Comment by GroovinChip** [comment]
Fair enough

**Comment by bdlukaa** [comment]
I have some ideas for this. We can have a `MouseCursor` property in the `MacosThemeData` class, and all the inputs will use it.

By default, on macOS and iOS, the cursor used will be `MouseCursor.defer`. On the other platforms, `SystemMouseCursors.click` shall be used. Something like:

```dart
MouseCursor _defaultMouseCursor() {
  switch (defaultTargetPlatform) {
    case TargetPlatform.macOS:
    case TargetPlatform.iOS:
      return MouseCursor.defer;
    default:
      return SystemMouseCursors.click;
  }
}
```

**Comment by wilsonowilson** [comment]
That sounds great!

**Comment by GroovinChip** [comment]
@wilsonowilson so to recap: we're saying that all mouse cursors should default to something like `MouseCursor.defer`, correct?

**Comment by wilsonowilson** [comment]
@GroovinChip Yeah, although I'm not sure if we'd want to use `MouseCursor.defer` over `SystemMouseCursors.basic` because it defers the cursor handling to its children.
